clients/drcachesim/tests/burst_traceopts.cpp

```diff
@@ -209,9 +209,53 @@ gather_trace(const std::string &tracer_ops, const std::string &out_subdir)
     return post_process(out_subdir);
 }
 
+static void
+reg_id_set_unit_tests()
+{
+    reg_id_set_t set;
+    assert(set.begin() == set.end());
+    // Test non-GPR.
+    auto insert_res = set.insert(IF_X86_ELSE(DR_REG_XMM0, DR_REG_Q0));
+    assert(insert_res.first == set.end());
+    assert(!insert_res.second);
+    assert(set.begin() == set.end());
+    // Test adding a GPR.
+    insert_res = set.insert(DR_REG_START_GPR + 1);
+    assert(insert_res.first != set.end());
+    assert(insert_res.second);
+    assert(set.begin() != set.end());
+    // Test find not-present.
+    auto find_res = set.find(DR_REG_START_GPR);
+    assert(find_res == set.end());
```

Comment:
Add an assert statement to show that `*find_res == DR_REG_INVALID` in this case. Adding a comment over `operator*` to document this will also be helpful.